cmd/gitserver/server/server.go

```
@@ -1214,6 +1215,23 @@ func (s *Server) cloneRepo(ctx context.Context, repo api.RepoName, opts *cloneOp
 		tmpPath = filepath.Join(tmpPath, ".git")
 		tmp := GitDir(tmpPath)
 
+		// It may already be cloned
+		if !repoCloned(dir) {
+			if err := s.setCloneStatus(ctx, repo, types.CloneStatusCloning); err != nil {
```

nit: if the callsite never changes behaviour due to error, why not not return error and instead log internally?

---

I don't want to hide the error and I'd rather the call site decides whether failing to save the error is fatal or not. Currently the call sites just log but I don't want to assume all future callers want that too.